source/vibe/web/rest.d

```diff
@@ -762,336 +613,253 @@ private HTTPServerRequestDelegate jsonMethodHandler(alias Func, size_t ridx, T)(
-
-			// legacy :id special case, left for backwards-compatibility reasons
-			static if (i == 0 && ParamNames[0] == "id") {
-				static if (is(PT == Json))
-					url_prefix = q{urlEncode(id.toString())~"/"};
-				else
-					url_prefix = q{urlEncode(toRestString(serializeToJson(id)))~"/"};
-			} else static if (anySatisfy!(mixin(CompareParamName.Name), WPAT)) {
-				alias PWPAT = Filter!(mixin(CompareParamName.Name), WPAT);
-				static if (PWPAT[0].origin == ParameterKind.header) {
-					// Don't send 'out' parameter, as they should be default init anyway and it might confuse some server
-					static if (!(PSCT[i] & PSC.out_)) {
-						param_handling_str ~= format(q{headers__["%s"] = to!string(%s);}, PWPAT[0].field, PWPAT[0].identifier);
-					}
-					static if (PSCT[i] & PSC.ref_ || PSCT[i] & PSC.out_) {
-						// Optional parameter
-						static if (isInstanceOf!(Nullable, PT)) {
-							param_handling_str ~= q{
-								optHdrs__["%2$s"] = null;
-								scope (exit)
-									%1$s = to!(TemplateArgsOf!(typeof(%1$s)))(
-										optHdrs__.get("%2$s", null));
-							}.format(ParamNames[i], PWPAT[0].field);
-						} else {
-							param_handling_str ~= q{
-								reqHdrs__["%2$s"] = null;
-								scope (exit)
-									if (auto ptr = "%2$s" in reqHdrs__)
-										%1$s = to!(typeof(%1$s))(*ptr);
-							}.format(ParamNames[i], PWPAT[0].field);
-						}
-					}
-				} else static if (PWPAT[0].origin == ParameterKind.query)
-					queryParamCTMap[PWPAT[0].field] = PWPAT[0].identifier;
-				else static if (PWPAT[0].origin == ParameterKind.body_)
-					bodyParamCTMap[PWPAT[0].field] = PWPAT[0].identifier;
-				else
-					static assert (0, "Internal error: Unknown ParameterKind in REST client code generation.");
-			} else static if (!ParamNames[i].startsWith("_")
-					  && !IsAttributedParameter!(Func, ParamNames[i])) {
-				// underscore parameters are sourced from the HTTPServerRequest.params map or from url itself
-				defaultParamCTMap[stripTUnderscore(ParamNames[i], null)] = ParamNames[i];
-			}
-		}
+		if (request_filter) request_filter(req);
 
-		// Block 3
-		string request_str;
-
-		static if (!meta.hadPathUDA) {
-			request_str = q{
-				if (m_settings.stripTrailingUnderscore && url__.endsWith("_"))
-					url__ = url__[0 .. $-1];
-				url__ = concatURL(%s, adjustMethodStyle(url__, m_methodStyle));
-			}.format(url_prefix);
-		} else {
-			import std.array : split;
-			auto parts = meta.url.split("/");
-			request_str ~= `url__ = ` ~ url_prefix;
-			foreach (i, p; parts) {
-				if (i > 0) {
-					request_str ~= `~ "/"`;
-				}
-				bool match = false;
-				if (p.startsWith(":")) {
-					foreach (pn; ParamNames) {
-						if (pn.startsWith("_") && p[1 .. $] == pn[1 .. $]) {
-							request_str ~= q{ ~ urlEncode(toRestString(serializeToJson(%s)))}.format(pn);
-							match = true;
-							break;
-						}
-					}
-				}
+		if (body_ != "")
+			req.writeBody(cast(ubyte[])body_, hdrs.get("Content-Type", "application/json"));
```

Comment:
I think it probably came from those compile fixes due to the temporary rejection of `if (!str)` for DMD master. Yeah, but I'm kinda used to `.length`, too.